v8.0.0-alpha.0 (#15319)
mui/mui-x: CHANGELOG.md

<!--/ HIGHLIGHT_ABOVE_SEPARATOR /-->

### Data Grid

#### `@mui/x-data-grid@8.0.0-alpha.0`

- [DataGrid] Fix resizing right pinned column (#15107) @KenanYusuf
- [DataGrid] Set default overlay height in flex parent layout (#15202) @cherniavskii
- [DataGrid] refactor: baseMenuList & baseMenuItem (#15049) @romgrk
- [DataGrid] refactor: remove more material imports (#15063) @romgrk
- [DataGrid] - Fix grid overlay aligment with scroll for rtl (#15072) @kalyan90
- [I10n] Improve Polish (pl-PL) locale (#15227) @belkocik

#### `@mui/x-data-grid-pro@8.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@8.0.0-alpha.0`, plus:

- [DataGridPro] Fix column pinning layout (#14966) @cherniavskii

#### `@mui/x-data-grid-premium@8.0.0-alpha.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@8.0.0-alpha.0`, plus:

- [DataGridPremium] Server-side data source with row grouping (#13826) @MBilalShafi

### Date and Time Pickers

#### Breaking changes

> - The default DOM structure of the field has changed. [Migration Guide - New DOM structure for the field](https://github.com/mui/mui-x/blob/master/docs/data/migration/migration-pickers-v7/migration-pickers-v7.md#new-dom-structure-for-the-field)

Author: My issue with that is that it gets too big for no added value. Add it to each of the migration url and you have a lot of useless info.

It would be fine in the docs or in a blog post though

> - The default DOM structure of the field has changed. [Migration Guide - New DOM structure for the field](https://github.com/mui/mui-x/blob/master/docs/data/migration/migration-pickers-v7/migration-pickers-v7.md#new-dom-structure-for-the-field)

Author: I used the gh file because it is more static, the `next.mui.com` will make this url useless very soon

> - The default DOM structure of the field has changed. [Migration Guide - New DOM structure for the field](https://github.com/mui/mui-x/blob/master/docs/data/migration/migration-pickers-v7/migration-pickers-v7.md#new-dom-structure-for-the-field)

Author: > > I used the gh file because it is more static, the next.mui.com will make this url useless very soon
> 
> Not sure to follow you here

Wouldn't `next.mui.com` be "stuck" after we release v8 until we release v9 next year? Or is `next` always up-to-date? 🤔

> - The default DOM structure of the field has changed. [Migration Guide - New DOM structure for the field](https://github.com/mui/mui-x/blob/master/docs/data/migration/migration-pickers-v7/migration-pickers-v7.md#new-dom-structure-for-the-field)

Reviewer: > Wouldn't next.mui.com be "stuck" after we release v8 until we release v9 next year? Or is next always up-to-date? 🤔

AFAIK, we will replace all the `next.mui` links with `mui.com` links when releasing major.

> - The default DOM structure of the field has changed. [Migration Guide - New DOM structure for the field](https://github.com/mui/mui-x/blob/master/docs/data/migration/migration-pickers-v7/migration-pickers-v7.md#new-dom-structure-for-the-field)

Reviewer 2: I confirm all links to the docs that appears somewhere else should be prefixed with `next.mui.com` and updated between version.

We had the same issue with JSDocs and agreed to use `https://mui.com` and replace them by `https://next.mui.com` according to the current status of the branch

> - The default DOM structure of the field has changed. [Migration Guide - New DOM structure for the field](https://github.com/mui/mui-x/blob/master/docs/data/migration/migration-pickers-v7/migration-pickers-v7.md#new-dom-structure-for-the-field)

Reviewer: Updated

#### `@mui/x-date-pickers@8.0.0-alpha.0`

- [fields] Enable the new field DOM structure by default (#14651) @flaviendelangle
- [fields] Remove `UseDateFieldComponentProps` and equivalent interfaces (#15053) @flaviendelangle
- [fields] Remove clear button from the tab sequence (#14616) @k-rajat19
- [l10n] Improve Greek (el-GR) locale (#15250) @GeorgiosDrivas
- [pickers] Clean definition of validation props (#15198) @flaviendelangle
- [pickers] Clean the new `ownerState` object (#15056) @flaviendelangle
- [pickers] Correctly type the `ownerState` of the `field` and `actionBar` slots when resolved in a picker component (#15162) @flaviendelangle
- [pickers] Fix `DateCalendar` timezone management (#12321) @LukasTy
- [pickers] Fix `DigitalClock` time options on a `DST` switch day (#10793) @LukasTy
- [pickers] Remove `TDate` generics in favor of `PickerValidDate` direct usage (#15001) @flaviendelangle
- [pickers] Remove `utils` and `value` params from translations (#14986) @arthurbalduini
- [pickers] Remove plural in "Pickers" on recently introduced APIs (#15297) @flaviendelangle

> - [pickers] Remove the re-export from `@mui/x-license` (#14487) @k-rajat19

Reviewer: ```suggestion
- [pickers] Remove the re-export from `@mui/x-license` (#14487) @k-rajat19
- [pickers] Unify JSDoc for all the disabled and readOnly props (#15304) @flaviendelangle 
```

# Changelog

> For full v7 changelog, please refer to the [v.7x branch](https://github.com/mui/mui-x/blob/v7.x/CHANGELOG.md).

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 8.0.0-alpha.0

> <img width="100%" alt="MUI X v8 Alpha is live" src="https://github.com/user-attachments/assets/114cf615-b617-435f-8499-76ac3c26c57b">

Author: ```suggestion
<img width="100%" alt="MUI X v8 Alpha is live" src="https://github.com/user-attachments/assets/114cf615-b617-435f-8499-76ac3c26c57b">
```

> _Nov 7, 2024_

Reviewer 2: ```suggestion
_Nov 14, 2024_
```

- The following types are no longer exported by `@mui/x-date-pickers` and `@mui/x-date-pickers-pro` — [Learn more](https://next.mui.com/x/migration/migration-pickers-v7/#removed-types).

  - `UseDateFieldComponentProps`
  - `UseTimeFieldComponentProps`
  - `UseDateTimeFieldComponentProps`
  - `BaseSingleInputFieldProps`
  - `BaseMultiInputFieldProps`
  - `BasePickersTextFieldProps`

- The `TDate` generic has been removed from all the types, interfaces, and variables of the `@mui/x-date-pickers` and `@mui/x-date-pickers-pro` packages — [Learn more](https://next.mui.com/x/migration/migration-pickers-v7/#remove-tdate-generic).

- The renamed `usePickersTranslation` and `usePickersContext` to have a coherent `Picker` prefix instead of `Pickers` — [Learn more](https://next.mui.com/x/migration/migration-pickers-v7/#renamed-variables).

> - The `LicenseInfo` object is not exported from the `@mui/x-date-pickers-pro` package anymore [Learn more](https://next.mui.com/x/migration/migration-pickers-v7/#stop-using-licenseinfo-from-mui-x-date-pickers-pro).

Reviewer: Could you add the improved wording to the migration guide as well? :pray:

- [pickers] Fix `DigitalClock` time options on a `DST` switch day (#10793) @LukasTy
- [pickers] Remove `TDate` generics in favor of `PickerValidDate` direct usage (#15001) @flaviendelangle
- [pickers] Remove `utils` and `value` params from translations (#14986) @arthurbalduini
- [pickers] Remove plural in "Pickers" on recently introduced APIs (#15297) @flaviendelangle
- [pickers] Remove the re-export from `@mui/x-license` (#14487) @k-rajat19
- [pickers] Strictly type the props a picker passes to its field, and migrate all the custom field demos accordingly (#15197) @flaviendelangle
- [pickers] Unify JSDoc for all the `disabled` and `readOnly` props (#15304) @flaviendelangle
- [pickers] Use the new `ownerState` in `DateCalendar`, `DateRangeCalendar`, `MonthCalendar` and `YearCalendar` (#15171) @flaviendelangle
- [pickers] Use the new `ownerState` in `usePickersLayout` and `useXXXPicker` (#14994) @flaviendelangle

#### `@mui/x-date-pickers-pro@8.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@8.0.0-alpha.0`.

### Charts

#### Breaking changes

> - The `legend` prop got removed. To pass props to the legend, use `slotProps={{ legend: { ... } }}` instead. This can be automatically done with the codemod as long as the `legend` prop does not come from a destructured object.[Migration Guide - Legend props propagation](https://next.mui.com/x/migration/migration-charts-v7/#legend-props-propagation-✅)

Reviewer: Oops I missed this one

```suggestion
- The `legend` prop got removed. To pass props to the legend, use `slotProps={{ legend: { ... } }}` instead. This can be automatically done with the codemod as long as the `legend` prop does not come from a destructured object — [Learn more](https://next.mui.com/x/migration/migration-charts-v7/#legend-props-propagation-✅).
```

> - The `legend` prop got removed. To pass props to the legend, use `slotProps={{ legend: { ... } }}` instead. This can be automatically done with the codemod as long as the `legend` prop does not come from a destructured object.[Migration Guide - Legend props propagation](https://next.mui.com/x/migration/migration-charts-v7/#legend-props-propagation-✅)

Reviewer 3: Not worth the effort to fix here, but in the future avoid "got removed"—this is a very informal phrasing, and it's unclear who or what did the removing. Instead, use "has been removed"—this implies that we, the maintainers, made the change.

> - The `legend` prop got removed. To pass props to the legend, use `slotProps={{ legend: { ... } }}` instead. This can be automatically done with the codemod as long as the `legend` prop does not come from a destructured object.[Migration Guide - Legend props propagation](https://next.mui.com/x/migration/migration-charts-v7/#legend-props-propagation-✅)

Reviewer: cc @alexfauquette

#### `@mui/x-charts@8.0.0-alpha.0`

- [charts] Introduce `hideLegend` prop (#15277) @alexfauquette
- [charts] Filter item outside the drawing area for perf (#14281) @alexfauquette
- [charts] Fix log scale with `null` data (#15337) @alexfauquette
- [charts] Fix tooltip follow mouse (#15189) @alexfauquette
- [charts] Remove `xAxisKey`, `yAxisKey`, and `zAxisKey` (#15192) @alexfauquette
- [charts] Remove axis from the pie chart (#15187) @alexfauquette
- [charts] Remove deprecated `legend` props (#15081) @alexfauquette
- [charts] Remove deprecated highlight properties (#15191) @alexfauquette
- [charts] Update Popper position outside of React (#15003) @alexfauquette
- [charts] Improve the performance of the `getSymbol` method (#15233) @romgrk

#### `@mui/x-charts-pro@8.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-charts@8.0.0-alpha.0`, plus:

> 
> - [charts-pro] Fix missing `typeOverload` (#15400) @alexfauquette

Reviewer 2: I 'm surprised it's here since the PR target v7.x https://github.com/mui/mui-x/pull/15400/files
```suggestion
```

Reviewer: My bad, I manually added all the missing PR and got it wrong on this one

- [DataGridPro] Toggle row expansion with `Enter` key in "tree data" (#15271) @k-rajat19

#### `@mui/x-data-grid-premium@8.0.0-alpha.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@8.0.0-alpha.0`, plus:

- [DataGridPremium] Fix incorrect rows selection count when selection propagation is enabled with row grouping (#15216) @arminmeh
- [DataGridPremium] Keep focus on the grouping cell on space bar press (#15108) @k-rajat19
- [DataGridPremium] Server-side data source with row grouping (#13826) @MBilalShafi

### Date and Time Pickers

#### Breaking changes

- The default DOM structure of the field has changed.
  Before version `v8.x`, the fields' DOM structure consisted of an `<input />`, which held the whole value for the component, but unfortunately presents a few limitations in terms of accessibility when managing multiple section values.
  Starting with version `v8.x`, all the field and picker components come with a new DOM structure that allows the field component to set aria attributes on individual sections, providing a far better experience with screen readers.

>   **Fallback to the non-accessible DOM structure:**

Reviewer: This section is empty without the code snippet :laughing: 

For the new DOM structure you can just remove everything under `**Fallback to the non-accessible DOM structure:**`.

And I would add a link to the right section of the migration guide for each breaking change that was shortened.
Something like:

See [Migration guide—New DOM structure for the field](https://next.mui.com/x/migration/migration-pickers-v7/#new-dom-structure-for-the-field) for more details.

Author: Added the migration guide to the changes: https://github.com/mui/mui-x/blob/186313072fd5714ccbadb95042f6a8ed7d5871a0/CHANGELOG.md

Had to manually add them though, and slightly reworked the text in some cases. 

This is not the latest version from master, I'll do another pass when we have a set release date

Reviewer: > Had to manually add them though, and slightly reworked the text in some cases.

I'll add them on by PR descriptions for the next BCs :+1:

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 8.0.0-alpha.0

<!-- generated comparing v7.22.1..master -->

_Nov 7, 2024_

We'd like to offer a big thanks to the 21 contributors who made this release possible. Here are some highlights ✨:

> TODO INSERT HIGHLIGHTS

Author: Is this necessary? A lot of packages have a lot of updates 😅

> TODO INSERT HIGHLIGHTS

Reviewer: ```suggestion
- 🔁 Support [automatic parents and children selection](https://next.mui.com/x/react-tree-view/rich-tree-view/selection/#automatic-parents-and-children-selection) for the Rich Tree View components.
```

> 

Author: Doesn't look like it generated the user list...

From https://github.com/mui/mui-x/pull/14913

> ### Tree View

Author: I thought the `## Changelog` ordered/categorised the changelogs. I was wrong 🙃 

This output is very confusing

Reviewer: Nop it just adds everything at the top :laughing: 
It would be a nice improvement

Reviewer: We should move them in their respective product line inside a `#### Breaking changes` before merging this PR :+1:

Author: moved

  Learn more about the anatomy of the Tree Items and the customization utilities provided on the [Tree Item Customization page](https://next.mui.com/x/react-tree-view/tree-item-customization/).

- The `onClick` and `onMouseDown` were the only event callback that were passed to the content of the Tree Item instead of its root.
  The goal was to make sure that the callback was not fired when clicking on a descendant of a giving item.
  This inconsistency has been solved, all the event manager now target the root of the item, and you can use the `onItemClick` prop on the Tree View component to target the content of an item:

  ```diff
  -<SimpleTreeView>
  +<SimpleTreeView onItemClick={handleItemClick}>
  -  <TreeItem onClick={handleItemClick}>
  +  <TreeItem >
   </SimpleTreeView>
  ```

- All the new Tree Item-related components and utils (introduced in the previous major to improve the DX of the Tree Item component) are becoming the default way of using the Tree Item and are therefore losing their `2` suffix:

  ```diff
   import * as React from 'react';
   import {
  -  TreeItem2,
  +  TreeItem,

>   -  TreeItem2Root,

Author: @flaviendelangle is all this information necessary? It feels like we are not going straight to the point and that these should be kept in the PR or migration guide.

Reviewer: So far AFAIK we always added the exact same content on the migration guide and on the changelog/release note.

If we want to stop doing it and instead add a summary + a link to the migration guide any time it's a longer section, I'm fine doing it.

WDYT @cherniavskii @alexfauquette @LukasTy @joserodolfofreitas?

Reviewer: I guess we could keep the "introduction text" as is and replace all the examples with a link to the migration guide.

Reviewer 4: I guess it's only for the `alpha.0` that the BCs take a bit more than expected space, from next versions, since there'll be a release every week, it shouldn't be as long.

Author: i've removed the examples, let me know if the wording is wrong or where we should keep the examples.

 I tried to proofread where I removed them, but you know better than I what was necessary or not

  + } from '@mui/x-tree-view/useTreeItem';
  - import { useTreeItem2Utils } from '@mui/x-tree-view/hooks';
  + import { useTreeItemUtils } from '@mui/x-tree-view/hooks';
   import {
  -  TreeItem2Provider,
  +  TreeItemProvider,
  -  TreeItem2ProviderProps,
  +  TreeItemProviderProps,
  - } from '@mui/x-tree-view/TreeItem2Provider';
  + } from '@mui/x-tree-view/TreeItemProvider';
   import {
  -  TreeItem2Icon,
  +  TreeItemIcon,
  -  TreeItem2IconProps,
  +  TreeItemIconProps,
  -  TreeItem2IconSlots,
  +  TreeItemIconSlots,
  -  TreeItem2IconSlotProps,
  +  TreeItemIconSlotProps,
  - } from '@mui/x-tree-view/TreeItem2Icon';
  + } from '@mui/x-tree-view/TreeItemIcon';
   import {
  -  TreeItem2DragAndDropOverlay,
  +  TreeItemDragAndDropOverlay,
  -  TreeItem2DragAndDropOverlayProps,
  +  TreeItemDragAndDropOverlayProps,
  - } from '@mui/x-tree-view/TreeItem2DragAndDropOverlay';
  + } from '@mui/x-tree-view/TreeItemDragAndDropOverlay';
   import {
  -  TreeItem2LabelInput,
  +  TreeItemLabelInput,
  -  TreeItem2LabelInputProps,
  +  TreeItemLabelInputProps,
  - } from '@mui/x-tree-view/TreeItem2LabelInput';
  + } from '@mui/x-tree-view/TreeItemLabelInput';
  ```

> From https://github.com/mui/mui-x/pull/14651

Reviewer: Yes it's generated automatically
But we need to move each BC description in its product line and remove thos "From XXX" :

> From https://github.com/mui/mui-x/pull/14651

Reviewer: In the past, what we did is add a `#### Breaking changes` at the beginning of each product line

For example:

````
### Date and Time Pickers

#### Breaking changes

- The `DesktopDateTimePicker` view rendering has been optimized by using the same technique as for `DesktopDateTimeRangePicker`.
  - The `dateTimeViewRenderers` have been removed in favor of reusing existing time view renderers (`renderTimeViewClock`, `renderDigitalClockTimeView` and `renderMultiSectionDigitalClockTimeView`) and date view renderer (`renderDateViewCalendar`).
  - Passing `renderTimeViewClock` to time view renderers will no longer revert to the old behavior of rendering only date or time view.

#### `@mui/x-date-pickers@7.0.0`

- [fields] Allow to override the separator between the start and the end date in all range fields (#12174) @flaviendelangle
- .... the rest
````

> From https://github.com/mui/mui-x/pull/14651

Author: I meant for the automatic PR description

> From https://github.com/mui/mui-x/pull/14651

Author: So we could use `## Changelog` here, and `## Migration` elsewhere

> From https://github.com/mui/mui-x/pull/14651

Reviewer: I'm not sure to follow what you mean by "automatic PR description"

> From https://github.com/mui/mui-x/pull/14651

Author: Oops. I mean that we add `## Changelog` and it generates the changelog from the PR description. We could have a `## Migration` which would be different from the changelog, to keep it simpler, and could be used to generate a migration doc in the future.

> From https://github.com/mui/mui-x/pull/14651

Reviewer 2: We could, the remaining question being "Will people use it?" 😈

But I'm probably too pessimistic on this aspect 😇

### Pickers

- The following variables were renamed to have a coherent `Picker` / `Pickers` prefix:

`usePickersTranslation`

```diff
- import { usePickersTranslation } from '@mui/x-date-pickers/hooks';
- import { usePickersTranslation } from '@mui/x-date-pickers';
- import { usePickersTranslation } from '@mui/x-date-pickers-pro';
+ import { usePickerTranslation } from '@mui/x-date-pickers/hooks';
+ import { usePickerTranslation } from '@mui/x-date-pickers';
+ import { usePickerTranslation } from '@mui/x-date-pickers-pro';
- const translations = usePickersTranslation();
+ const translations = usePickerTranslation();
```

`usePickersContext`

```diff
- import { usePickersContext } from '@mui/x-date-pickers/hooks';
- import { usePickersContext } from '@mui/x-date-pickers';
- import { usePickersContext } from '@mui/x-date-pickers-pro';
+ import { usePickerContext } from '@mui/x-date-pickers/hooks';
+ import { usePickerContext } from '@mui/x-date-pickers';
+ import { usePickerContext } from '@mui/x-date-pickers-pro';
- const pickersContext = usePickersContext();
+ const pickerContext = usePickerContext();
```

From https://github.com/mui/mui-x/pull/14487

> - [pickers] The `LicenseInfo` is not exported anymore from `@mui/x-date-pickers-pro`. Import it from `@mui/x-license` instead.

Reviewer: ````suggestion
- The `LicenseInfo` object is not exported from the `@mui/x-date-pickers-pro` package anymore.
You can import it from `@mui/x-license` instead:

  ```diff
  -import { LicenseInfo } from '@mui/x-date-pickers-pro';
  +import { LicenseInfo } from '@mui/x-license';
  
  LicenseInfo.setLicenseKey('YOUR_LICENSE_KEY');
  ```
````

- [DataGrid] Prevent error when deleting the last row (#15146) @dpak-maurya
- [DataGrid] Set default overlay height in flex parent layout (#15202) @cherniavskii
- [DataGrid] refactor: baseMenuList & baseMenuItem (#15049) @romgrk
- [DataGrid] refactor: remove more material imports (#15063) @romgrk
- [I10n][DataGrid] Improve Polish (pl-PL) locale (#15227) @belkocik

#### `@mui/x-data-grid-pro@8.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@8.0.0-alpha.0`, plus:

- [DataGridPro] Add list view tests (#15043) @KenanYusuf
- [DataGridPro] Fix column pinning layout (#14966) @cherniavskii
- [DataGridPro] Should toggle row expansion with Enter key in Tree data (#15271) @k-rajat19

#### `@mui/x-data-grid-premium@8.0.0-alpha.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@8.0.0-alpha.0`, plus:

- [DataGridPremium] Fix incorrect rows selection count when selection propagation is enabled with row grouping (#15216) @arminmeh
- [DataGridPremium] Keep focus on the grouping cell on space bar press (#15108) @k-rajat19

> - [DataGridPremium] Server-side data source with row grouping (#13826) @MBilalShafi

Author: I had to manually remove them 🙃

#### `@mui/x-date-pickers-pro@8.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@8.0.0-alpha.0`.

### Charts

#### `@mui/x-charts@8.0.0-alpha.0`

- [charts] Allow `SeriesValueFormatter` to return `null` value (#15057) @clins1994
- [charts] Fix tooltip follow mouse (#15189) @alexfauquette
- [charts] Remove `xAxisKey`, `yAxisKey`, and `zAxisKey` (#15192) @alexfauquette
- [charts] Remove axis from the pie chart (#15187) @alexfauquette
- [charts] Remove deprecated `legend` props (#15081) @alexfauquette
- [charts] Remove deprecated highlight properties (#15191) @alexfauquette
- [charts] Update Popper position outside of react (#15003) @alexfauquette
- [charts] perf: getSymbol (#15233) @romgrk

#### `@mui/x-charts-pro@8.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-charts@8.0.0-alpha.0`.

### Tree View

#### `@mui/x-tree-view@8.0.0-alpha.0`

- [TreeView] Always apply the indentation on the item content instead of its parent's group (#15089) @flaviendelangle

> - [TreeView] Automatic parents and children selection (#14899) @flaviendelangle

Reviewer: Done

- [TreeView] Fix drag and drop color usage (#15042) @LukasTy
- [TreeView] Remove deprecated `TreeView` component (#15093) @flaviendelangle
- [TreeView] Replace `<TreeItem />` with `<TreeItem2 />` and migrate all the components and utils (#14913) @flaviendelangle
- [TreeView] Replace the Tree Item anatomy images (#15066) @noraleonte

### Docs

> - [docs] Add section explaining how to keep the selection while filtering (#15185) @arminmeh

Reviewer: ```suggestion
- [docs] Add migration guide for the removal of `LicenseInfo` from `@mui/x-date-pickers-pro` (#15321) @flaviendelangle 
- [docs] Add section explaining how to keep the selection while filtering (#15185) @arminmeh
```

- [docs] Apply the new DX to the Button Field demos (#14860) @flaviendelangle
- [docs] Apply the new DX to the `Autocomplete` Field demo (#15165) @flaviendelangle
- [docs] Cleanup the pickers migration guide (#15310) @flaviendelangle
- [docs] Copyedit the Charts Getting Started sequence (#14962) @samuelsycamore
- [docs] Create Pickers masked field recipe (#13515) @flaviendelangle
- [docs] Fix missing punctuation on descriptions (#15229) @oliviertassinari
- [docs] Fix peer dependency range (#15281) @oliviertassinari
- [docs] Fix the `AdapterMomentHijri` doc section (#15312) @flaviendelangle
- [docs] Start v8 migration guides (#15096) @MBilalShafi
- [docs] Subdivide and reorganize navigation bar (#15014) @samuelsycamore
- [docs] Use `PickersTextField` in the customization playground (#15288) @LukasTy
- [docs] Use `next` instead of `^8.0.0` in the migration guides (#15091) @flaviendelangle

### Core

> - [core] Adjust cherry-pick GH actions (#15099) @LukasTy

Reviewer: ```suggestion
- [core] Adjust the `cherry-pick` GitHub actions (#15099) @LukasTy
```

> - [core] Apply () function convention (#15075) @oliviertassinari

Reviewer: ```suggestion
- [core] Add `()` at the name of function name in the doc (#15075) @oliviertassinari
```

- [core] Fix CodeSandbox and StackBlitz for next doc-infra sync @oliviertassinari
- [core] Fix Vale error on master @oliviertassinari
- [core] Fix changelog reference to VoiceOver @oliviertassinari
- [core] Fix tools-public.mui.com redirection @oliviertassinari
- [core] Move `helpers` to `@mui/x-internals` package (#15188) @LukasTy

> - Fix grid overlay aligment with scroll for rtl (#15072) @kalyan90

Reviewer: Should be in the grid section

> - [code-infra] Set renovate to automerge devDependencies (#13463) @JCQuintas

Reviewer: ```suggestion
- [code-infra] Set renovate to automerge `devDependencies` (#13463) @JCQuintas
```

  **Apply custom indentation:**

  If you used to set custom indentation in your Tree Item, you can use the new `itemChildrenIndentation` prop to do it while supporting the new DOM structure:

  ```tsx
  <RichTreeView
    items={MUI_X_PRODUCTS}
    itemChildrenIndentation={24}
    defaultExpandedItems={['grid']}
  />
  ```

  See [Tree Item Customization—Change nested item's indentation](https://next.mui.com/x/react-tree-view/tree-item-customization/#change-nested-items-indentation) for more details.

  **Fallback to the old behavior:**

  If you used to style your content element (for example to add a border to it) and you don't use the drag and drop re-ordering, you can manually put the padding on the group transition element to restore the previous behavior:

  ```tsx
  const CustomTreeItemContent = styled(TreeItemContent)(({ theme }) => ({
    // Remove the additional padding of nested elements
    padding: theme.spacing(0.5, 1),
  }));

  const CustomTreeItemGroupTransition = styled(TreeItemGroupTransition)({
    // Add the padding back on the group transition element
    paddingLeft: 'var(--TreeView-itemChildrenIndentation) !important',
  });
  ```

> From https://github.com/mui/mui-x/pull/15191
> 
> The `highlighted` and `faded` properties of highlightScope were deprecated in favor of `highlight` and `fade`.
> The deprecated ones are now removed.
> From https://github.com/mui/mui-x/pull/15192
> 
> The `xAxisKey`, `yAxisKey`, and `zAxisKey` properties got deprecated in favor of `xAxisId`, `yAxisId`, and `zAxisId`.
> Those deprecated properties are now removed.

Reviewer 2: ```suggestion
```

+ import { usePickerTranslation } from '@mui/x-date-pickers/hooks';
+ import { usePickerTranslation } from '@mui/x-date-pickers';
+ import { usePickerTranslation } from '@mui/x-date-pickers-pro';
- const translations = usePickersTranslation();
+ const translations = usePickerTranslation();
```

`usePickersContext`

```diff
- import { usePickersContext } from '@mui/x-date-pickers/hooks';
- import { usePickersContext } from '@mui/x-date-pickers';
- import { usePickersContext } from '@mui/x-date-pickers-pro';
+ import { usePickerContext } from '@mui/x-date-pickers/hooks';
+ import { usePickerContext } from '@mui/x-date-pickers';
+ import { usePickerContext } from '@mui/x-date-pickers-pro';
- const pickersContext = usePickersContext();
+ const pickerContext = usePickerContext();
```

From https://github.com/mui/mui-x/pull/14487

- The `LicenseInfo` object is not exported from the `@mui/x-date-pickers-pro` package anymore.
  You can import it from `@mui/x-license` instead:

  ```diff
  -import { LicenseInfo } from '@mui/x-date-pickers-pro';
  +import { LicenseInfo } from '@mui/x-license';
  
   LicenseInfo.setLicenseKey('YOUR_LICENSE_KEY');
  ```

> From https://github.com/mui/mui-x/pull/15187
> 
> - The Pie Chart lost all props and render linked to axes because pie chart does not need cartesian axes. If you used it, you can still add them back with composition. Please consider opening an issue to share your use case with us.

Reviewer 2: ```suggestion
```

>   From https://github.com/mui/mui-x/pull/15081
> 
> - The `legend` prop got removed. To pass props to the legend, use `slotProps={{ legend: { ... } }}` instead. This can be automatically done with the codemod as long as the `legend` prop does not come from a destructured object.
> 
>   ```jsx
>   // Supported by the codemod
>   <LineChart legend={{ ... }} />
> 
>   // Not supported by the codemod
>   const props = { legend: { ... }, ... }
>   <Line {...props} />
>   ```
> 
> - The `slots.legend` does not receive the `drawingArea` prop. You can still access your custom legend with the `useDrawingArea()` hook if your custom legend needs it.

Reviewer 2: ```suggestion
```

- [pickers] Clean the new `ownerState` object (#15056) @flaviendelangle
- [pickers] Correctly type the `ownerState` of the `field` and `actionBar` slots when resolved in a picker component (#15162) @flaviendelangle
- [pickers] Fix `DateCalendar` timezone management (#12321) @LukasTy
- [pickers] Fix `DigitalClock` time options on a `DST` switch day (#10793) @LukasTy
- [pickers] Remove `TDate` generics in favor of `PickerValidDate` direct usage (#15001) @flaviendelangle
- [pickers] Remove `utils` and `value` params from translations (#14986) @arthurbalduini
- [pickers] Remove plural in "Pickers" on recently introduced APIs (#15297) @flaviendelangle
- [pickers] Remove the re-export from `@mui/x-license` (#14487) @k-rajat19
- [pickers] Use the new `ownerState` in `DateCalendar`, `DateRangeCalendar`, `MonthCalendar` and `YearCalendar` (#15171) @flaviendelangle
- [pickers] Use the new `ownerState` in `usePickersLayout` and `useXXXPicker` (#14994) @flaviendelangle

#### `@mui/x-date-pickers-pro@8.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@8.0.0-alpha.0`.

> ### Charts
> 

Reviewer 2: `````suggestion
### Charts

#### Breaking changes

- The `legend` prop got removed. To pass props to the legend, use `slotProps={{ legend: { ... } }}` instead. This can be automatically done with the codemod as long as the `legend` prop does not come from a destructured object.

  ```jsx
  // Supported by the codemod
  <LineChart legend={{ ... }} />

  // Not supported by the codemod
  const props = { legend: { ... }, ... }
  <Line {...props} />
  ```
  :::
  If you find `slotProps={{ legend: { hidden: true } }}` too long, you can now use the `hideLegend` prop to prevent rendering the legend.
  :::
- The `slots.legend` does not receive the `drawingArea` prop. You can still access your custom legend with the `useDrawingArea()` hook if your custom legend needs it.

`````

> #### `@mui/x-charts@8.0.0-alpha.0`

Reviewer 2: ```suggestion
- The `highlighted` and `faded` properties of highlightScope were deprecated in favor of `highlight` and `fade`.
The deprecated ones are now removed.
- The `xAxisKey`, `yAxisKey`, and `zAxisKey` properties got deprecated in favor of `xAxisId`, `yAxisId`, and `zAxisId`.
- The Pie Chart lost all props and renderer linked to axes because pie chart does not need cartesian axes. If you used it, you can still add them back with composition. Please consider opening an issue to share your use case with us.

#### `@mui/x-charts@8.0.0-alpha.0`
```

> - [charts] Allow `SeriesValueFormatter` to return `null` value (#15057) @clins1994

Reviewer 2: ```suggestion
- [charts] Introduce `hideLegend` prop (#15277) @alexfauquette
- [charts] Allow `SeriesValueFormatter` to return `null` value (#15057) @clins1994
```

- [charts] Remove axis from the pie chart (#15187) @alexfauquette
- [charts] Remove deprecated `legend` props (#15081) @alexfauquette
- [charts] Remove deprecated highlight properties (#15191) @alexfauquette
- [charts] Update Popper position outside of React (#15003) @alexfauquette
- [charts] Improve the performance of the `getSymbol` method  (#15233) @romgrk

#### `@mui/x-charts-pro@8.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-charts@8.0.0-alpha.0`.

### Tree View

#### `@mui/x-tree-view@8.0.0-alpha.0`

- [TreeView] Always apply the indentation on the item content instead of its parent's group (#15089) @flaviendelangle
- [TreeView] Automatic parents and children selection (#14899) @flaviendelangle
- [TreeView] Fix drag and drop color usage (#15042) @LukasTy
- [TreeView] Remove deprecated `TreeView` component (#15093) @flaviendelangle
- [TreeView] Replace `<TreeItem />` with `<TreeItem2 />` and migrate all the components and utils (#14913) @flaviendelangle
- [TreeView] Replace the Tree Item anatomy images (#15066) @noraleonte

### Docs

> - [docs] Add migration guide for the removal of `LicenseInfo` from `@mui/x-date-pickers-pro` (#15321) @flaviendelangle 

Reviewer 2: ```suggestion
- [docs] Add migration guide for the removal of `LicenseInfo` from `@mui/x-date-pickers-pro` (#15321) @flaviendelangle 
- [docs] Add migration guide for the first breaking changes of charts (#15276) @alexfauquette
```
